### PR TITLE
SAP, SAR, POA&M schematron and xspec bare bones

### LINKED ISSUE #458: Scaffold SAP, SAR, POA&M Schematron
**Extended Description**
- As a developer, in order to have a framework to place SAP, SAR, and POA&M validations, I would like placeholder Schematron documents for each to exist.

**Acceptance Criteria**
- [ ] `sap.sch` with one basic assertion exists
- [ ] `sap.xspec` with validation of `sap.sch`
- [ ] `sar.sch` with one basic assertion exists
- [ ] `sar.xspec` with validation of `sar.sch`
- [ ] `poam.sch` with one basic assertion exists
- [ ] `poam.xspec` with validation of `poam.sch` exists
- [ ] SCH and XSPEC integrated into build system in manner analogous to SSP validations
- [ ] All Schematron assertion messages are declarative statements which affirm the positive test outcome.
- [ ] All Schematron assertion diagnostic messages are declarative statements which explain the negative test outcome.
- [ ] The Schematron code has no assertion failures when validated using `src/validations/styleguides/sch.sch` using the `basic` phase.
- [ ] XSpec unit tests for positive and negative Schematron assertion outcomes accompany all Schematron assertions (where feasible).
- More acceptance criteria…

**Story Tasks**
- [ ] Tasks…

**Definition of Done**
 - [ ] Acceptance criteria met
 - [ ] Unit test coverage of our code > 95%  
 - [ ] Automated code quality checks passed 
 - [ ] Security reviewed and reported
 - [ ] Reviewed against plain language guidelines
 - [ ] Code must be self-documenting
 - [ ] No local tech debt
 - [ ] Load/performance tests passed – needs to be created/automated
 - [ ] Documentation updated
 - [ ] Architectural Decision Record completed as necessary for significant design choices
 - [ ] PR reviewed & approved
 - [ ] Source code merged

### PR DESCRIPTION
**Pull request by markXLIX**
Created SAP, SAR, POAM schematron and xspec starter files.
Added matching results.html files to gitignore.

### CODE REVIEW

File: src/validations/rules/poam.sch [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    </sch:phase>
+
+    <doc:xspec
+        href="../test/poam.xspec" />
+    
+    <sch:title>FedRAMP Plan of Action and Milestones Validations</sch:title>
+    <xsl:output
+        encoding="UTF-8"
+        indent="yes"
+        method="xml" />
```
Comment (GaryGapinski) [inline src/validations/rules/poam.sch:33]: Why is there XSL code here?

File: src/validations/rules/sap.sch [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    </sch:phase>
+
+    <doc:xspec
+        href="../test/sap.xspec" />
+    
+    <sch:title>FedRAMP Security Assessment Plan Validations</sch:title>
+    <xsl:output
+        encoding="UTF-8"
+        indent="yes"
+        method="xml" />
```
Comment (GaryGapinski) [inline src/validations/rules/sap.sch:33]: Why is there XSL code here?

File: src/validations/rules/sar.sch [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    </sch:phase>
+
+    <doc:xspec
+        href="../test/sar.xspec" />
+    
+    <sch:title>FedRAMP Security Assessment Results Validations</sch:title>
+    <xsl:output
+        encoding="UTF-8"
+        indent="yes"
+        method="xml" />
```
Comment (GaryGapinski) [inline src/validations/rules/sar.sch:33]: Why is there XSL code here?

File: src/validations/rules/poam.sch [hunk 4]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="utf-8"?>
+<?xml-model schematypens="http://purl.oclc.org/dsdl/schematron" href="../styleguides/sch.sch" phase="basic" title="Schematron Style Guide for FedRAMP Validations" ?>
+<sch:schema
+    queryBinding="xslt2"
+    xmlns:doc="https://fedramp.gov/oscal/fedramp-automation-documentation"
+    xmlns:oscal="http://csrc.nist.gov/ns/oscal/1.0"
+    xmlns:sch="http://purl.oclc.org/dsdl/schematron"
```
Comment (GaryGapinski) [inline src/validations/rules/poam.sch:7]: The `xmlns:unit="http://us.gov/testing/unit-testing"` namespace is missing. You recently created it. Why would you not include it?

File: src/validations/rules/poam.sch [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    <sch:ns
+        prefix="oscal"
+        uri="http://csrc.nist.gov/ns/oscal/1.0" />
+    <sch:ns
+        prefix="doc"
+        uri="https://fedramp.gov/oscal/fedramp-automation-documentation" />
+    <sch:ns
+        prefix="fedramp"
+        uri="https://fedramp.gov/ns/oscal" />
+
```
Comment (GaryGapinski) [inline src/validations/rules/poam.sch:19]: The
```
    <sch:ns
        prefix="unit"
        uri="http://us.gov/testing/unit-testing" />
```
namespace is missing. You recently created it. Why would you not include it?

File: src/validations/rules/sap.sch [hunk 6]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="utf-8"?>
+<?xml-model schematypens="http://purl.oclc.org/dsdl/schematron" href="../styleguides/sch.sch" phase="basic" title="Schematron Style Guide for FedRAMP Validations" ?>
+<sch:schema
+    queryBinding="xslt2"
+    xmlns:doc="https://fedramp.gov/oscal/fedramp-automation-documentation"
+    xmlns:oscal="http://csrc.nist.gov/ns/oscal/1.0"
+    xmlns:sch="http://purl.oclc.org/dsdl/schematron"
```
Comment (GaryGapinski) [inline src/validations/rules/sap.sch:7]: The `xmlns:unit="http://us.gov/testing/unit-testing"` namespace is missing. You recently created it. Why would you not include it?

File: src/validations/rules/sar.sch [hunk 7]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="utf-8"?>
+<?xml-model schematypens="http://purl.oclc.org/dsdl/schematron" href="../styleguides/sch.sch" phase="basic" title="Schematron Style Guide for FedRAMP Validations" ?>
+<sch:schema
+    queryBinding="xslt2"
+    xmlns:doc="https://fedramp.gov/oscal/fedramp-automation-documentation"
+    xmlns:oscal="http://csrc.nist.gov/ns/oscal/1.0"
+    xmlns:sch="http://purl.oclc.org/dsdl/schematron"
```
Comment (GaryGapinski) [inline src/validations/rules/sar.sch:7]: The `xmlns:unit="http://us.gov/testing/unit-testing"` namespace is missing. You recently created it. Why would you not include it?

File: src/validations/rules/sap.sch [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    <sch:ns
+        prefix="oscal"
+        uri="http://csrc.nist.gov/ns/oscal/1.0" />
+    <sch:ns
+        prefix="doc"
+        uri="https://fedramp.gov/oscal/fedramp-automation-documentation" />
+    <sch:ns
+        prefix="fedramp"
+        uri="https://fedramp.gov/ns/oscal" />
+
```
Comment (GaryGapinski) [inline src/validations/rules/sap.sch:19]: The
```
    <sch:ns
        prefix="unit"
        uri="http://us.gov/testing/unit-testing" />
```
namespace is missing. You recently created it. Why would you not include it?

File: src/validations/rules/sar.sch [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    <sch:ns
+        prefix="oscal"
+        uri="http://csrc.nist.gov/ns/oscal/1.0" />
+    <sch:ns
+        prefix="doc"
+        uri="https://fedramp.gov/oscal/fedramp-automation-documentation" />
+    <sch:ns
+        prefix="fedramp"
+        uri="https://fedramp.gov/ns/oscal" />
+
```
Comment (GaryGapinski) [inline src/validations/rules/sar.sch:19]: The
```
    <sch:ns
        prefix="unit"
        uri="http://us.gov/testing/unit-testing" />
```
namespace is missing. You recently created it. Why would you not include it?

File: src/validations/test/poam.xspec [hunk 10]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<x:description
+    schematron="../rules/poam.sch"
+    xmlns:doc="https://fedramp.gov/oscal/fedramp-automation-documentation"
+    xmlns:sch="http://purl.oclc.org/dsdl/schematron"
+    xmlns:sqf="http://www.schematron-quickfix.com/validator/process"
```
Comment (GaryGapinski) [inline src/validations/test/poam.xspec:7]: `http://www.schematron-quickfix.com/validator/process` is a Schematron QuickFix (SQF)-related namespace. It is used to declare Schematron quick fix constructs in a **Schematron document**. This is an XSpec document. Why is SQF declared here?

File: src/validations/test/sap.xspec [hunk 11]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<x:description
+    schematron="../rules/sap.sch"
+    xmlns:doc="https://fedramp.gov/oscal/fedramp-automation-documentation"
+    xmlns:sch="http://purl.oclc.org/dsdl/schematron"
+    xmlns:sqf="http://www.schematron-quickfix.com/validator/process"
```
Comment (GaryGapinski) [inline src/validations/test/sap.xspec:7]: `http://www.schematron-quickfix.com/validator/process` is a Schematron QuickFix (SQF)-related namespace. It is used to declare Schematron quick fix constructs in a **Schematron document**. This is an XSpec document. Why is SQF declared here?

File: src/validations/test/sar.xspec [hunk 12]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<x:description
+    schematron="../rules/sar.sch"
+    xmlns:doc="https://fedramp.gov/oscal/fedramp-automation-documentation"
+    xmlns:sch="http://purl.oclc.org/dsdl/schematron"
+    xmlns:sqf="http://www.schematron-quickfix.com/validator/process"
```
Comment (GaryGapinski) [inline src/validations/test/sar.xspec:7]: `http://www.schematron-quickfix.com/validator/process` is a Schematron QuickFix (SQF)-related namespace. It is used to declare Schematron quick fix constructs in a **Schematron document**. This is an XSpec document. Why is SQF declared here?

File: src/validations/rules/poam.sch [hunk 13]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="utf-8"?>
+<?xml-model schematypens="http://purl.oclc.org/dsdl/schematron" href="../styleguides/sch.sch" phase="basic" title="Schematron Style Guide for FedRAMP Validations" ?>
+<sch:schema
+    queryBinding="xslt2"
+    xmlns:doc="https://fedramp.gov/oscal/fedramp-automation-documentation"
+    xmlns:oscal="http://csrc.nist.gov/ns/oscal/1.0"
+    xmlns:sch="http://purl.oclc.org/dsdl/schematron"
+    xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
```
Comment (GaryGapinski) [inline src/validations/rules/poam.sch:8]: Why is this here?

File: src/validations/rules/sap.sch [hunk 14]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="utf-8"?>
+<?xml-model schematypens="http://purl.oclc.org/dsdl/schematron" href="../styleguides/sch.sch" phase="basic" title="Schematron Style Guide for FedRAMP Validations" ?>
+<sch:schema
+    queryBinding="xslt2"
+    xmlns:doc="https://fedramp.gov/oscal/fedramp-automation-documentation"
+    xmlns:oscal="http://csrc.nist.gov/ns/oscal/1.0"
+    xmlns:sch="http://purl.oclc.org/dsdl/schematron"
+    xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
```
Comment (GaryGapinski) [inline src/validations/rules/sap.sch:8]: Why is this here?

File: src/validations/rules/sar.sch [hunk 15]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="utf-8"?>
+<?xml-model schematypens="http://purl.oclc.org/dsdl/schematron" href="../styleguides/sch.sch" phase="basic" title="Schematron Style Guide for FedRAMP Validations" ?>
+<sch:schema
+    queryBinding="xslt2"
+    xmlns:doc="https://fedramp.gov/oscal/fedramp-automation-documentation"
+    xmlns:oscal="http://csrc.nist.gov/ns/oscal/1.0"
+    xmlns:sch="http://purl.oclc.org/dsdl/schematron"
+    xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
```
Comment (GaryGapinski) [inline src/validations/rules/sar.sch:8]: Why is this here?

**Review by GaryGapinski — COMMENTED** [review]
You have chosen to exclude the following namespaces from Schematron documents which are in active use in the `ssp.sch` Schematron document:
```
    xmlns:array="http://www.w3.org/2005/xpath-functions/array"
    xmlns:feddoc="http://us.gov/documentation/federal-documentation"
    xmlns:map="http://www.w3.org/2005/xpath-functions/map"
    xmlns:unit="http://us.gov/testing/unit-testing"
```
If that exclusion was deliberate, what was the rationale?

**Comment by danielnaab** [comment]
@markXLIX @GaryGapinski The Makefile should be applying the styleguides and xspec to all appropriate documents. Apparently, previously some of this was not getting applied correctly. There may also have been some that failed silently. As a result, there are a few failing tests/styleguides on this branch.

**Review by danielnaab — APPROVED** [review]
